### PR TITLE
HV-1617 Make our pom files more consistent with WildFly and JBoss parent

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1617
 * https://hibernate.atlassian.net/browse/HV-1627
 * https://hibernate.atlassian.net/browse/HV-1626

Not the most interesting PR but all our projects are now consistent in the naming and more in line with what is done in WF parent and JBoss parent.

This should help for productization.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -95,131 +95,186 @@
...
-        <!-- URLs used in javadoc and documentation generation -->
-        <bv.spec.url>http://beanvalidation.org/2.0/spec/</bv.spec.url>
-        <java.api-docs.base-url>http://docs.oracle.com/javase/8/docs/api</java.api-docs.base-url>
-        <java.technotes.base-url>http://docs.oracle.com/javase/8/docs/technotes</java.technotes.base-url>
-        <javaee.api-docs.base-url>http://docs.oracle.com/javaee/7/api</javaee.api-docs.base-url>
-        <javafx.docs.url>http://docs.oracle.com/javase/8/javase-clienttechnologies.htm</javafx.docs.url>
-        <bv.api-docs.base-url>http://docs.jboss.org/hibernate/beanvalidation/spec/2.0/api</bv.api-docs.base-url>
-        <javamoney.api-docs.base-url>http://javamoney.github.io/apidocs</javamoney.api-docs.base-url>
+        <!-- JVM options required for Arquillian and WildFly -->
+        <arquillian.jvm.args></arquillian.jvm.args>
```
Comment (yrodiere) [inline pom.xml]: Note that Arquillian can be used with much more than WildFly, so it may be wise to at least put the term "wildfly" in there if you intend to put wildfly-specific JVM options.

File: pom.xml [hunk 2]
```diff
@@ -95,131 +95,186 @@
...
-        <arquillian.javaVmArguments.add-opens></arquillian.javaVmArguments.add-opens>
+        <maven.compiler.argument.target>${maven.compiler.target}</maven.compiler.argument.target>
+        <maven.compiler.argument.source>${maven.compiler.source}</maven.compiler.argument.source>
+        <maven.compiler.argument.testTarget>${maven.compiler.testTarget}</maven.compiler.argument.testTarget>
+        <maven.compiler.argument.testSource>${maven.compiler.testSource}</maven.compiler.argument.testSource>
 
-        <!-- Forbidden API related properties -->
-        <forbiddenapis-junit.path>forbidden-junit.txt</forbiddenapis-junit.path>
+        <!-- Set empty default values to avoid Maven leaving property references (${...}) when it doesn't find a value -->
+        <surefire.jvm.args></surefire.jvm.args>
```
Comment (yrodiere) [inline pom.xml]: So, to explain the rationale... In Search, I used "jvm.args" for properties containing arguments that are specific to a version of the JVM, and "system.args" for the actual arguments passed to surefire/failsafe. Yeah, it's dumb.

If you don't use the exact same property names as Search, and thus are not constrained by consistency with Search, I would recommend to use clearer names, like what I proposed above.

File: pom.xml [hunk 3]
```diff
@@ -414,38 +469,34 @@
...
     </dependencyManagement>
 
     <build>
-        <extensions>
-            <extension>
-                <groupId>org.apache.maven.wagon</groupId>
-                <artifactId>wagon-webdav</artifactId>
-                <version>1.0-beta-2</version>
-            </extension>
-        </extensions>
```
Comment (yrodiere) [inline pom.xml]: That looks dubious. Maybe move it to a separate commit with an explanation?

File: pom.xml [hunk 4]
```diff
@@ -746,63 +801,50 @@
...
                 </plugin>
-                <plugin>
-                    <groupId>org.codehaus.mojo</groupId>
-                    <artifactId>exec-maven-plugin</artifactId>
-                    <!--
-                    1.6.0 has a bug that affects us, waiting for the next version
-                    https://github.com/mojohaus/exec-maven-plugin/issues/75
-                    -->
-                    <version>1.5.0</version>
-                </plugin>
```
Comment (yrodiere) [inline pom.xml]: Same here. Separate commit?

File: pom.xml [hunk 5]
```diff
@@ -746,63 +801,50 @@
...
                     </dependencies>
                 </plugin>
                 <plugin>
                     <groupId>ch.mfrey.maven.plugin</groupId>
                     <artifactId>copy-maven-plugin</artifactId>
-                    <version>0.0.6</version>
-                </plugin>
-                <plugin>
-                    <artifactId>maven-project-info-reports-plugin</artifactId>
-                    <version>2.9</version>
```
Comment (yrodiere) [inline pom.xml]: Same here. Separate commit?

File: pom.xml [hunk 6]
```diff
@@ -1140,12 +1175,11 @@
                 <jdk>[9,)</jdk>
             </activation>
             <properties>
-                <maven-surefire-plugin.argLine.add-modules>--add-modules=${maven-surefire-plugin.jigsaw.modules}</maven-surefire-plugin.argLine.add-modules>
```
Comment (yrodiere) [inline pom.xml]: Is the add-modules option not necessary anymore?

File: tck-runner/pom.xml [hunk 7]
```diff
@@ -47,8 +47,8 @@
             <artifactId>javax.el</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.hibernate.javax.persistence</groupId>
-            <artifactId>hibernate-jpa-2.1-api</artifactId>
+            <groupId>javax.persistence</groupId>
+            <artifactId>javax.persistence-api</artifactId>
```
Comment (yrodiere) [inline tck-runner/pom.xml]: Same as above:

> Might want a separate commit for that? That has way more implications that just refactoring the properties.

File: cdi/pom.xml [hunk 8]
```diff
@@ -214,7 +214,7 @@
             </activation>
             <properties>
                 <!-- required by Weld -->
-                <maven-surefire-plugin.argLine.add-opens>--add-opens java.base/java.lang=ALL-UNNAMED</maven-surefire-plugin.argLine.add-opens>
+                <surefire.jvm.add-opens>--add-opens java.base/java.lang=ALL-UNNAMED</surefire.jvm.add-opens>
```
[the remark this answers is not in the document]
Comment (gsmet) [reply cdi/pom.xml]: To be honest, I'm not sure. For now, I decided to keep it that way as the intent is clear. If at some point we need other things for future versions, I'll revisit.

I don't think it's easy to take the right decision completely in the dark and I fear we will lose the clarity we have now.

File: pom.xml (hunk 6 above)
Comment (gsmet) [reply pom.xml]: Yes, we don't need it anymore as we removed our JAXB dependency: as it's totally removed from JDK 11, we needed to move to Stax (or add a specific dependency but that's not what we decided).

File: documentation/pom.xml [hunk 9]
```diff
@@ -40,7 +40,7 @@
 
         <!-- Skip artifact deployment -->
         <maven.deploy.skip>true</maven.deploy.skip>
-        <maven-surefire-plugin.argLine>-Duser.language=en</maven-surefire-plugin.argLine>
+        <surefire.jvm.args>-Duser.language=en</surefire.jvm.args>
```
[the remark this answers is not in the document]
Comment (gsmet) [reply documentation/pom.xml]: Yes, that's what it did before so I'll keep it that way for now. Either we'll need a separate property or define the locale globally. Not decided for now.

File: pom.xml [hunk 10]
```diff
@@ -95,131 +95,186 @@
...
+        <version.license.plugin>3.0</version.license.plugin>
+        <version.release.plugin>2.5.3</version.release.plugin>
+        <version.resources.plugin>3.0.2</version.resources.plugin>
+        <version.shade.plugin>3.1.0</version.shade.plugin>
+        <version.sigtest.plugin>1.0</version.sigtest.plugin>
+        <version.source.plugin>3.0.1</version.source.plugin>
+        <version.surefire.plugin>2.21.0</version.surefire.plugin>
+        <version.failsafe.plugin>${version.surefire.plugin}</version.failsafe.plugin>
+        <version.wildfly.plugin>1.2.1.Final</version.wildfly.plugin>
+        <version.org.wildfly.core>5.0.0.Final</version.org.wildfly.core>
```
[the remark this answers is not in the document]
Comment (gsmet) [reply pom.xml]: I decided it to put it there as it's tightly related to the WildFly Maven plugin. I did the same thing in OGM.

File: distribution/pom.xml [hunk 11]
```diff
@@ -54,8 +54,8 @@
             <artifactId>log4j</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.hibernate.javax.persistence</groupId>
-            <artifactId>hibernate-jpa-2.1-api</artifactId>
+            <groupId>javax.persistence</groupId>
+            <artifactId>javax.persistence-api</artifactId>
```
[the remark this answers is not in the document]
Comment (gsmet) [reply distribution/pom.xml]: Done.

File: engine/pom.xml [hunk 12]
```diff
@@ -279,7 +279,7 @@
                 <jdk>[9,)</jdk>
             </activation>
             <properties>
-                <maven-surefire-plugin.argLine>--illegal-access=deny</maven-surefire-plugin.argLine>
+                <surefire.jvm.args>--illegal-access=deny</surefire.jvm.args>
```
[the remark this answers is not in the document]
Comment (gsmet) [reply engine/pom.xml]: I improved the situation slightly.

File: pom.xml (hunk 1 above)
Comment (gsmet) [reply pom.xml]: Yeah, I had an hesitation here. Looks like you had the same so let's add `wildfly`.

File: pom.xml (hunk 3 above)
Comment (gsmet) [reply pom.xml]: Done.

File: pom.xml (hunk 4 above)
Comment (gsmet) [reply pom.xml]: Done.

File: pom.xml (hunk 5 above)
Comment (gsmet) [reply pom.xml]: Done.

**Comment by gsmet** [comment]
@yrodiere force-pushed an update that should address your concerns. The changes are in a fixup commit.

File: modules/pom.xml [hunk 13]
```diff
@@ -314,7 +315,7 @@
                 <jdk>[9,)</jdk>
             </activation>
             <properties>
-                <patch-gen.system.args>--add-opens=java.base/java.lang=ALL-UNNAMED</patch-gen.system.args>
+                <patch-gen.jvm.args.add-opens>--add-opens=java.base/java.lang=ALL-UNNAMED</patch-gen.jvm.args.add-opens>
```
Comment (yrodiere) [inline modules/pom.xml]: Looks like this is never used? You're probably missing a default value for `patch-gen.jvm.args`.

**Comment by gsmet** [comment]
Merged! Thanks for the review.
